WordPress/Classes/Stores/PluginStore.swift

```diff
@@ -217,6 +273,38 @@ private extension PluginStore {
         }
     }
 
+    func fetchPluginDirectoryEntries(site: JetpackSiteRef) {
+        state.plugins[site]?.plugins
+            .map({ $0.slug })
+            .filter(shouldFetchDirectory(slug:))
+            .forEach(fetchPluginDirectoryEntry(slug:))
+    }
+
+    func fetchPluginDirectoryEntry(slug: String) {
+        let remote = PluginDirectoryServiceRemote()
+        state.fetchingDirectoryEntry[slug] = true
+        remote.getPluginInformation(
+            slug: slug,
+            success: { [actionDispatcher] (entry) in
+                actionDispatcher.dispatch(PluginAction.receivePluginDirectoryEntry(slug: slug, entry: entry))
+            },
+            failure: { [actionDispatcher] (error) in
+                actionDispatcher.dispatch(PluginAction.receivePluginDirectoryEntryFailed(slug: slug, error: error))
+        })
+    }
+
+    func receivePluginDirectoryEntry(slug: String, entry: PluginDirectoryEntry) {
+        transaction { (state) in
+            state.directoryEntries[slug] = .present(entry)
+            state.fetchingDirectoryEntry[slug] = false
+        }
+    }
+
+    func receivePluginDirectoryEntryFailed(slug: String, error: Error) {
+        // TODO: mark entry as missing if not found
```

Comment:
I still have to do this so we don't try to fetch info for plugins that we know are not on the directory 😬